### PR TITLE
Divyanshu | Test | Version 3 - Get all voucher page

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cupmt5p

### CODE REVIEW

File: apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.ts [hunk 1]
```diff
@@ -325,7 +325,6 @@ export class ContentFilterComponent implements DoCheck, OnInit, OnChanges, OnDes
      */
     public changeInvoiceHeader(event: boolean): void {
         this.customTemplate.sections['header'].data['formNameInvoice'].display = event;
-        this.onChangeFieldVisibility(null, null, null);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.ts:328]: Checked and Confirmed with @dilpreetsinghdang  sir

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 2]
```diff
@@ -790,7 +788,7 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
                 if (isElectron) {
                     this.router.navigate(['/login']);
                 } else {
-                    window.location.href = (environment.production) ? `https://stage.giddh.com/login` : `https://test.giddh.com/login`;
+                    window.location.href = (environment.production) ? `https://giddh.com/login` : `https://test.giddh.com/login`;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/shared/header/header.component.ts:791]: Need to check !!!!

Comment (dvCodeWorld) [reply apps/web-giddh/src/app/shared/header/header.component.ts:791]: Whole file

File: apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts [hunk 3]
```diff
@@ -1,15 +1,77 @@
...
 })
 export class ConvertBillDialogComponent implements OnInit {
+    /* This will hold local JSON data */
+    @Input() public localeData: any = {};
+    /* This will hold common JSON data */
+    @Input() public commonLocaleData: any = {};
+    /** Emits convert To Bill form value */
+    @Output() public convertToBill = new EventEmitter<any>();
+    /** Hold Form group */
+    public form: FormGroup;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts]: Need to change this !!!

File: apps/web-giddh/src/app/shared/header/header.component.ts (hunk 2 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/shared/header/header.component.ts:791]: Please change url stage and test to giddh in whole file

File: apps/web-giddh/src/app/login/login.component.html [hunk 4]
```diff
@@ -191,7 +191,7 @@ <h4 class="mb-4">Login to your secure accounting space</h4>
                                             >
                                         </li>
                                     </ul>
-                                    <ul class="social-login" *ngIf="showAppleLogin">
+                                    <!-- <ul class="social-login" *ngIf="showAppleLogin">
                                         <li class="google-ico">
                                             <a href="javascript:void(0);" (click)="appleLogin()">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/login/login.component.html:196]: Confirmed with Kriti we are not showing apple login button

File: apps/web-giddh/src/app/new-inventory/component/create-new-inventory-component/create-new-inventory.component.html [hunk 5]
```diff
@@ -4,7 +4,7 @@
         <label class="page-title-inventory"> Create Inventory </label>
 
         <div class="inventory-header-right">
-            <button mat-icon-button class="ml-3" style="font-size: 25px" [routerLink]="['/pages/new-inventory']">
+            <button mat-icon-button class="ml-3" style="font-size: 25px" [routerLink]="['/pages/inventory/v2']">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/create-new-inventory-component/create-new-inventory.component.html:7]: Ye html use nh ho rha h but fr bh humne url update rkhi h

File: apps/web-giddh/src/app/services/http-wrapper.service.ts [hunk 6]
```diff
@@ -75,7 +75,7 @@ export class HttpWrapperService {
         options.headers["Session-Id"] = this.generalService.sessionId;
         options.headers["Content-Type"] = "application/json";
         options.headers["Accept"] = "application/json";
-        options.headers["X-Tenant"] = this.generalService.getUtmParameter("X-Tenant");
+        // options.headers["X-Tenant"] = this.generalService.getUtmParameter("X-Tenant");
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/services/http-wrapper.service.ts:78]: Informed to Leena for handle this if this on Api if task is not going to prod.

File: apps/web-giddh/src/app/shared/header/header.component.ts [hunk 7]
```diff
@@ -1487,17 +1484,11 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
             } else if (
                 document.getElementsByTagName("tabset") &&
                 document.getElementsByTagName("tabset").length > 0 &&
-                !this.router.url.includes("/vendor") && (!document.getElementsByClassName("static-tabs-on-page") || (document.getElementsByClassName("static-tabs-on-page") && document.getElementsByClassName("static-tabs-on-page").length === 0))) {
+                !this.router.url.includes("/vendor")) {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/header.component.ts]: revert

File: apps/web-giddh/src/app/shared/primary-sidebar/primary-sidebar.component.ts [hunk 8]
```diff
@@ -499,7 +499,6 @@ export class PrimarySidebarComponent implements OnInit, OnChanges, OnDestroy {
             item.time = +new Date();
             let entity = (item.type) === 'MENU' ? 'menus' : 'accounts';
             this.doEntryInDb(entity, item, fromInvalidState);
-            this.closeAccountModal(true);
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/primary-sidebar/primary-sidebar.component.ts]: revert

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/shared/primary-sidebar/primary-sidebar.component.ts (hunk 8 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/shared/primary-sidebar/primary-sidebar.component.ts]: Done

**Review by dilpreetsinghofficial — APPROVED** [review]

File: apps/web-giddh/src/app/shared/header/header.component.ts (hunk 7 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/shared/header/header.component.ts]: Done

File: apps/web-giddh/src/app/services/http-wrapper.service.ts (hunk 6 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/services/http-wrapper.service.ts:78]: Done

File: apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts (hunk 3 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts]: Done

File: apps/web-giddh/src/app/reports/components/advance-receipt-report/advance-receipt-report.component.ts [hunk 9]
```diff
@@ -790,9 +789,9 @@ export class AdvanceReceiptReportComponent implements AfterViewInit, OnDestroy,
      * @memberof AdvanceReceiptReportComponent
      */
     public previewVoucher(receipt: any): void {
-        if (this.voucherApiVersion === 2) {
-            this.router.navigate(['/pages/voucher/receipt/preview/' + receipt.uniqueName + '/' + receipt.account?.uniqueName]);
-        }
+        // if (this.voucherApiVersion === 2) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/reports/components/advance-receipt-report/advance-receipt-report.component.ts]: confirm with kriti

File: apps/web-giddh/src/app/reports/components/payment-report/payment-report.component.html [hunk 10]
```diff
@@ -15,14 +15,14 @@
             >
                 <!-- Invoice filter section -->
                 <div class="mr-r1 d-flex flex-sm-row flex-column">
-                    <button
+                    <!-- <button
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/reports/components/payment-report/payment-report.component.html]: confirm with kriti

File: apps/web-giddh/src/app/reports/components/payment-report/payment-report.component.ts [hunk 11]
```diff
@@ -690,9 +690,9 @@ export class PaymentReportComponent implements AfterViewInit, OnDestroy, OnInit
      * @memberof PaymentReportComponent
      */
     public previewVoucher(payment: any): void {
-        if (this.voucherApiVersion === 2) {
-            this.router.navigate(['/pages/voucher/payment/preview/' + payment.uniqueName + '/' + payment.account?.uniqueName]);
-        }
+        // if (this.voucherApiVersion === 2) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/reports/components/payment-report/payment-report.component.ts]: confirm with kriti

File: apps/web-giddh/src/assets/css/style-1.scss [hunk 12]
```diff
@@ -3465,15 +3412,6 @@ td.mat-footer-cell {
         }
     }
 }
-.aside-pane {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/css/style-1.scss:3468]: check this

File: apps/web-giddh/src/assets/css/style-1.scss [hunk 13]
```diff
@@ -3195,58 +3194,6 @@ td.mat-footer-cell {
...
-                    letter-spacing: 0px;
-                    font: normal normal 600 16px/22px Open Sans;
-                    color: var(--color-tab-primary-active) !important;
-                }
-            }
-        }
-    }
-}
-
-.mat-chip-list-wrapper {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/css/style-1.scss:3245]: check this

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 14]
```diff
@@ -11043,37 +11038,6 @@ body.update-scroll-hidden {
     }
 }
 
-.ledger-attachments-popup {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/assets/css/style-2.scss]: check this

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.html [hunk 15]
```diff
@@ -1,148 +1,245 @@
...
+                <mat-option [value]="localeData?.e_invoice_statuses_label?.yet_to_be_pushed">{{ localeData?.e_invoice_statuses_label?.yet_to_be_pushed }}</mat-option>
+                <mat-option [value]="localeData?.e_invoice_statuses_label?.pushed">{{ localeData?.e_invoice_statuses_label?.pushed }}</mat-option>
+                <mat-option [value]="localeData?.e_invoice_statuses_label?.push_initiated">{{ localeData?.e_invoice_statuses_label?.push_initiated }}</mat-option>
+                <mat-option [value]="localeData?.e_invoice_statuses_label?.cancelled">{{ localeData?.e_invoice_statuses_label?.cancelled }}</mat-option>
+                <mat-option [value]="localeData?.e_invoice_statuses_label?.mark_as_cancelled">{{ localeData?.e_invoice_statuses_label?.mark_as_cancelled }}</mat-option>
+                <mat-option [value]="localeData?.e_invoice_statuses_label?.failed">{{ localeData?.e_invoice_statuses_label?.failed }}</mat-option>
+                <mat-option [value]="localeData?.e_invoice_statuses_label?.na">{{ localeData?.e_invoice_statuses_label?.na }}</mat-option>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.html:58]: Check this

File: apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.html [hunk 16]
```diff
@@ -1,148 +1,245 @@
...
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.html:93]: check this

File: apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.html [hunk 17]
```diff
@@ -1,148 +1,245 @@
...
+
+    <ng-container *ngIf="type === 'proforma'">
+        <div class="mt-3">
+            <div class="d-flex column-gap1">
+                <div class="custom-mat-form-field">
+                    <div class="date-range-wrapper">
+                        <mat-form-field
+                            [matDatepickerToggleIcon]="false"
+                            appearance="outline"
+                            (click)="picker.open()"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.html]:
```suggestion
                            (click)="picker?.open()"
```

File: apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.html [hunk 18]
```diff
@@ -1,148 +1,245 @@
...
+    </ng-container>
+
+    <ng-container *ngIf="type === 'purchase-order'">
+        <div class="d-flex d-flex column-gap1 mr-b15 w-100">
+            <div class="custom-mat-form-field w-50">
+                <div class="date-range-wrapper">
+                    <mat-form-field
+                        [matDatepickerToggleIcon]="false"
+                        appearance="outline"
+                        (click)="picker.open()"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.html]:
```suggestion
                        (click)="picker?.open()"
```

File: apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.html [hunk 19]
```diff
@@ -1,148 +1,245 @@
...
+    </ng-container>
+
     <!-- purchase-order-search-start -->
-    <div class="mt-3">
-      <label>Expected Delivery Date</label>
-      <div class="custom-datepicker-wrapper mt-1 mr-b15">
-        <div class="custom-datepicker">
-            <input type="text" name="selectedDateRange"
-              class="giddh-datepicker date-range-picker" />
+    <!-- <div class="mt-3">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/advance-search/advance-search.component.html]: remove commented code

File: apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts [hunk 20]
```diff
@@ -1,15 +1,77 @@
...
+        public dialogRef: MatDialogRef<any>,
+        private formBuilder: FormBuilder
+    ) {
+        this.initForm();
+    }
 
-  ngOnInit() {
-  }
+    /**
+     * Lifecycle hook for destroy
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/convert-bill-dialog/convert-bill-dialog.component.ts]: Correct documentation

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 21]
```diff
@@ -1,636 +1,758 @@
...
-                                            {{element.position}}
+                                        <div class="invoice-number" *ngIf="!selectedVouchers?.length">
+                                            {{ i+1 }}
                                         </div>
-                                        <div class="checkbox-hide">
+                                        <div [ngClass]="{'checkbox-hide': !selectedVouchers?.length }">
                                             <mat-checkbox
                                                 color="primary"
+                                                [checked]="isVoucherSelected(element)"
+                                                [value]="element.uniqueName"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/list/list.component.html:71]:
```suggestion
                                                [value]="element?.uniqueName"
```

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 22]
```diff
@@ -1,636 +1,758 @@
...
                                         ></ng-container>
-                                        Estimate No.
                                     </div>
-                                    </th>
+                                </th>
                                 <td mat-cell *matCellDef="let element">
-                                    <a href="javascript:;" class="btn-link-2">
-                                        {{element.estimate}}
-                                    </a>
+                                    <a href="javascript:;" class="btn-link-2">{{ element.voucherNumber ? element.voucherNumber : commonLocaleData?.app_not_available }}</a>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]:
```suggestion
                                    <a href="javascript:;" class="btn-link-2">{{ element?.voucherNumber ? element.voucherNumber : commonLocaleData?.app_not_available }}</a>
```

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 23]
```diff
@@ -1,636 +1,758 @@
...
+                                    <span class="d-block" *ngIf="element.expiredDays !== null">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]:
```suggestion
                                    <span class="d-block" *ngIf="element?.expiredDays !== null">
```

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 24]
```diff
@@ -1,636 +1,758 @@
...
-                                        <div class="p-3 border-bottom account-header">
-                                            <div class="d-flex align-items-center justify-content-between column-gap15 ">
-                                                <h3>00000000</h3>
-                                                <span class="cp">
-                                                    <img src="./assets/images/path.svg" />
-                                                </span>
-                                            </div>
-                                            <p class="mt-2">919111525164</p>
-                                            <p>ravinder@walkover.in</p>
+                                    <div class="cursor-pointer" (dblclick)="element.showAccountDetails = true">{{ (element.account?.customerName || element?.account?.name) | titlecase }}</div>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/list/list.component.html:122]:
```suggestion
                                    <div class="cursor-pointer" (dblclick)="element?.showAccountDetails = true">{{ (element.account?.customerName || element?.account?.name) | titlecase }}</div>
```

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 25]
```diff
@@ -1,636 +1,758 @@
...
+                                    <mat-card class="outline-card account-detail-wrapper" *ngIf="element.showAccountDetails" (clickOutside)="element.showAccountDetails = false">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: check nulity check in element

File: apps/web-giddh/src/app/vouchers/list/list.component.html [hunk 26]
```diff
@@ -1,636 +1,758 @@
...
                             </ng-container>
                             <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
                             <tr mat-row *matRowDef="let row; columns: displayedColumns;" class="checkbox-show"></tr>
                         </table>
-                        <mat-paginator [pageSizeOptions]="[5, 10, 20]"
-                            showFirstLastButtons
-                            aria-label="Select page of periodic elements">
-                        </mat-paginator>
+                        <mat-paginator
+                                *ngIf="!(getVouchersInProgress$ | async) && dataSource?.length && totalResults > pageSizeOptions[0]"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/list/list.component.html]: format code

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.html [hunk 27]
```diff
@@ -1,122 +1,71 @@
...
+            <span class="text-light d-inline-block">Base Currency</span>
+            <span class="mx-1">{{ commonLocaleData?.app_numbers?.one }} <ng-container *ngIf="currencySwitched">{{
+                    company?.baseCurrency }}</ng-container><ng-container *ngIf="!currencySwitched">{{
+                    voucherDetails?.account?.currency?.code }}</ng-container></span>
+            <span>
+                <img class="switch-icon middle mr-1 ml-1" src="assets/images/switch-icon.svg" alt="switch-icon"
+                    (click)="switchCurrency()" />
+            </span>
+            <span class="mx-1">
+                <img src="assets/images/edit-pencilicon.svg" class="cp mr-1" (click)="showExchangeRateEditField = true"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.html]:
```suggestion
                <img src="assets/images/edit-pencilicon.svg" class="cursor-pointer mr-1" (click)="showExchangeRateEditField = true"
```

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.html [hunk 28]
```diff
@@ -1,122 +1,71 @@
...
+                    *ngIf="!showExchangeRateEditField" />
+                <img src=" assets/images/autosave.svg" class="cp mr-1" (click)="showExchangeRateEditField = false"
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.html]:
```suggestion
                <img src=" assets/images/autosave.svg" class="cursor-pointer mr-1" (click)="showExchangeRateEditField = false"
```

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts [hunk 29]
```diff
@@ -1,15 +1,254 @@
...
+    public isMulticurrencyAccount = false;
+    /** Selected payment mode */
+    public selectedPaymentMode: any;
+    /** Holds true if Bank Selected */
+    public isBankSelected: boolean = false;
+    /** True if currency switched */
+    private currencySwitched: boolean = false;
+    /** True if we need to show exchange rate edit field */
+    public showExchangeRateEditField: boolean = false;
+    /** Holds true if save is inprogress */
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts]: Correct documentation

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts [hunk 30]
```diff
@@ -1,15 +1,254 @@
...
+        private formBuilder: FormBuilder
+    ) {
 
-  ngOnInit() {
-  }
+    }
 
-}
+    /**
+     * Lifecycle hook for destroy
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts]: Correct documentation

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts [hunk 31]
```diff
@@ -1,15 +1,254 @@
...
+            tagUniqueName: [''],
+            chequeNumber: [''],
+            chequeClearanceDate: [''],
+            description: [''],
+            exchangeRate: [1],
+            uniqueName: [this.voucherDetails?.uniqueName]
+        });
+
+        this.isMulticurrencyAccount = this.voucherDetails?.accountCurrencySymbol !== this.voucherDetails?.companyCurrencySymbol;
+
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts]: remove spacing

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts [hunk 32]
```diff
@@ -1,15 +1,254 @@
...
+        });
+
+        /** Exchange rate */
+        this.componentStore.exchangeRate$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                this.paymentForm.get('exchangeRate')?.patchValue(response);
+            }
+        });
+
+        this.settingsTagService.GetAllTags().pipe(takeUntil(this.destroyed$)).subscribe(response => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts:120]: move to component store or confirm with @ravinderwalkover sir to skip now

File: apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts [hunk 33]
```diff
@@ -1,15 +1,254 @@
...
+    public onSelectPaymentMode(event: any): void {
+        if (event && event.value) {
+            if (!this.isMulticurrencyAccount || this.voucherDetails?.account?.currency?.code === event?.additional?.currency) {
+                this.assignAmount(this.voucherDetails?.balanceDue?.amountForAccount, this.voucherDetails?.account?.currency?.symbol);
+            } else {
+                this.assignAmount(this.voucherDetails?.balanceDue?.amountForCompany, event?.additional?.currencySymbol);
+            }
+            this.selectedPaymentMode = event;
+
+            this.searchService.loadDetails(event.value).pipe(takeUntil(this.destroyed$)).subscribe(response => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/payment-dialog/payment-dialog.component.ts:160]: move to component store or confirm with @ravinderwalkover sir to skip now

File: apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts [hunk 34]
```diff
@@ -75,8 +91,22 @@ const DEFAULT_STATE: VoucherState = {
...
+    bulkUpdateVoucherInProgress: null,
+    bulkUpdateVoucherIsSuccess: null,
+    bulkExportVoucherInProgress: null,
+    bulkExportVoucherResponse: null,
+    actionVoucherIsSuccess: null,
+    adjustVoucherIsSuccess: null,
+    uploadImageBase64InProgress: false,
+    uploadImageBase64Response: null,
+    convertToInvoice: null,
+    convertToProforma: null,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts]:
```suggestion
    convertToProforma: null
```

File: apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts [hunk 35]
```diff
@@ -730,6 +783,32 @@ export class VoucherComponentStore extends ComponentStore<VoucherState> {
         );
     });
 
+    readonly getPurchaseOrders = this.effect((data: Observable<{ request: any }>) => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/utility/vouchers.store.ts:789]: add documentation after deployed on test on all files

File: apps/web-giddh/src/app/vouchers/utility/vouchers.utility.service.ts [hunk 36]
```diff
@@ -348,13 +359,17 @@ export class VouchersUtilityService {
...
 
-        invoiceForm = cleaner?.clean(invoiceForm, {
-            nullCleaner: true
-        });
+        invoiceForm = this.cleanObject(invoiceForm);
 
         return invoiceForm;
     }
 
+    public cleanObject(object: any): any {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vouchers/utility/vouchers.utility.service.ts:374]: add documentation

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

**Review by dilpreetsinghofficial — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**